backend/core/database.py

```diff
@@ -589,32 +590,33 @@ def handle_hijack_update(self, message):
                     rekey_update_keys = list(msg_["monitor_keys"])
                     rekey_updates = []
                     try:
-                        query = (
-                            "SELECT key, prefix, origin_as, peer_asn, as_path, service, "
-                            "type, communities, timestamp FROM bgp_updates "
-                            "WHERE bgp_updates.handled=false AND bgp_updates.key = %s"
-                        )
-
-                        with get_ro_cursor(self.ro_conn) as db_cur:
-                            psycopg2.extras.execute_batch(
-                                db_cur, query, (rekey_update_keys,), page_size=1000
+                        rekey_update_keys_lists = chunk_list(rekey_update_keys, 1000)
+                        for rekey_update_keys_list in rekey_update_keys_lists:
+                            query = (
+                                "SELECT key, prefix, origin_as, peer_asn, as_path, service, "
+                                "type, communities, timestamp FROM bgp_updates "
+                                "WHERE bgp_updates.handled=false AND bgp_updates.key IN "
+                                + str(tuple(rekey_update_keys_list))
```

Comment:
have you tested this? If it works, I totally agree with this approach.

Reply:
yes, it works. However, codefactor does not agree. I am trying to find a workaround to make codefactor happy.